pkg/network/payload/headers.go

```diff
@@ -19,6 +21,8 @@ const (
 func (p *Headers) DecodeBinary(br *io.BinReader) {
 	lenHeaders := br.ReadVarUint()
 
+	p.Received = lenHeaders
```

Comment:
What if we're to use `br.Err` to pass it as an error? I think it would still be possible to recover in `handleConn()`. I just don't like introducing artificial fields in a structure that is supposed to represent a network message payload.

Reply:
If we will do it via `br.Err` then we should expect expect error everytime we use `DecodeBinary` on a message containing `Headers`. I thought this could lead to some easy-to-miss bugs.
Maybe this is not a problem, though as `Headers` is used only as a last field of a single message.

Reply:
Well, we only have one point where we decode this message, if we're to have more then maybe they'll need to have different behaviour wrt this error. Even if not, passing it as an error makes more sense to me anyway.